skrifa/src/outline/memory.rs

```diff
@@ -0,0 +1,25 @@
+//! Support for temporary memory allocation, making use of the stack for
+//! small sizes.
+
+/// Invokes the callback with a memory buffer of the requested size.
+pub(super) fn with_temporary_memory<R>(size: usize, mut f: impl FnMut(&mut [u8]) -> R) -> R {
+    // Wrap in a function and prevent inlining to avoid stack allocation
+    // and zeroing if we don't take this code path.
```

Comment:
as in dead code analysis would not prune the branch if it was inlined? Is this a common trick?

Reply:
yeah, this just guarantees that we'll never end up with a stack frame that is the sum of all potential buffer sizes. LLVM would _probably_ do a dynamic stack adjustment (`alloca` style) for each branch but you never know...